src/Coordination/KeeperServer.cpp

```diff
@@ -388,10 +408,13 @@ void KeeperServer::shutdownRaftServer()
 
     raft_instance.reset();
 
-    if (asio_listener)
+    for (const auto & asio_listener : asio_listeners)
```

Comment:
It'd be more consistent to close listeners before `raft_instance.reset()` i.e. uninitialize it in reverse order. Otherwise, it can happen that we'll accept a connection but the object which should serve it will be already destroyed

Reply:
See https://github.com/ClickHouse/NuRaft/pull/45#discussion_r946691743